modules/service/src/main/scala/lucuma/odb/service/DatasetService.scala

```diff
@@ -30,6 +31,14 @@ import Services.Syntax.*
 
 sealed trait DatasetService[F[_]] {
 
+  /**
+   * Finds the dataset id consistent with the given ids (if any).
+   */
+  def resolveDid(
+    did: Option[Dataset.Id],
+    ref: Option[DatasetReference]
+  ): F[Result[Dataset.Id]]
```

Comment:
I thought I'd need this but then it turned out I didn't.  Still, it seems inevitable and it prompted generalizing the id resolver which is now shared with the `ObservationService` so I decided I'd leave it.